src/lib.rs

```diff
@@ -675,6 +720,41 @@ impl<K: KVStore + Sync + Send + 'static> Node<K> {
 		self.runtime.read().unwrap().is_some()
 	}
 
+	async fn create_channel_from_pj(
+		pj_req: PayjoinRequest, _payjoin_executer: Arc<PayjoinExecuter<K>>,
+	) {
+		let psbt = pj_req.clone().original_psbt();
+		let pj_response = PayjoinResponse::new(psbt);
+		// Send OpenChannel message
+		// Wait for AcceptChannel message
+		// Send FundingCreated message
+		// Wait for FundingSigned message
+		// Build PayjoinResponse
+		// Send PayjoinResponse to queue
+		pj_req.clone().queue(pj_response);
+	}
+
+	/// Request a new channel to be opened with a remote peer.
+	pub fn payjoin_channel(
```

Comment:
this function is currently doing two things which is not great.
it should be broke into two functions and in the "payjoin_uri" docs we should mention that the `pending_channels` array should be filled first